docs/proposals/automated-node-removal.md

```diff
@@ -0,0 +1,60 @@
+# Proposal: Automation of node removal procedure
+
+Last updated: 24.06.2021
+
+## Abstract
+
+Implementation of the option to delete CSI CRs from a node, which going to be removed. 
+
+## Background
+
+User manual actions for node removal are described in [doc](https://github.com/dell/csi-baremetal/blob/master/docs/node-removal.md). The goal of this POC is automating most of the steps of that procedure. 
+
+## Proposal
+
+CSI CRs deletion(Volumes, AC, ACR, Drives, LVGs) will be triggerred on deletion of appropriate csibmnode and we be performed by node-controller.
+Before implementation, we need to replace node-uuid annotation with label and add node-affinity `uuid label is exists` to csi-baremetal-node daemonset.
+
+Updated node removal steps performed by user:
+1. Get node uuid mapped to node name
+2. Delete all related Pods and PVCs
+3. Delete csibmnode CR with same node uuid
+
+Node removal steps performed by node-controller:
+1. Check csibmnode deletion timestamp and start node removal if needed
+2. Delete Volume CRs and LVGs (patch finalizers if stacked)
+3. Remove uuid label from node
+4. Wait until csi-baremetal-node pod will be destroyed (to not restore Drive, LVG and AC CRs)
+5. Delete Drive, LVG and AC CRs
+
+The approach may work if k8sNode is a part of the cluster yet or if it is deleted (skip 3,4 steps).
```

Comment:
What is reject node removal process and criteria?

Reply:
Created `ISSUE-1` with description of the problem. Also, added note-removal process label to check if it Operator waits pre-conditions or rejects procedure